### ISSUE TITLE
Specification & OTEP relationship: Provide references to implemented OTEPs 

### ISSUE THREAD

**Issue by svrnm** [open-telemetry/opentelemetry-specification#3792]
**What are you trying to achieve?**

Please take a look at this issue by @hterik: https://github.com/open-telemetry/opentelemetry.io/issues/3685, the particular issue is that it is not clear from the Spec on [Span Status](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status)), what the intend of this section is. Only after going into the _Git Blame View_, digging up a related [PR](https://github.com/open-telemetry/opentelemetry-specification/pull/1685) and [issue](https://github.com/open-telemetry/opentelemetry-specification/issues/1584), I was able to find the [OTEP 136](https://github.com/open-telemetry/oteps/blob/main/text/trace/0136-error_flagging.md) which provides examples & background on _why_ things are specified this way.

This is only one instance of an issue I had a few times now: while reading the spec, I missed the full picture of where things are coming from and why they are the way they are. Knowing how things work, I looked for the right OTEP to learn more. Outsiders lack this knowledge.

Therefore, what I want to suggest with this issue, is that the specification has in-place back links to the OTEPs that have been implemented by a certain section. Ideally as close as possible to the content, e.g. in the case of `Span Status` this shouldn't be a Reference section at the very bottom of the page, but a footnote at the title, or a line at the beginning stating that this is implementing this and that OTEP.

Many OTEPs contain a lot of valuable contextual details, so I think this would help people a lot to understand the specification better by having them accessible from the right spot in the spec.

**Additional context.**

So far I could identify the following places where a reference is provided:

* https://github.com/open-telemetry/opentelemetry-specification/blob/3d60131181a6d4a82e1b7684c609e8528e86d9d2/specification/schemas/README.md?plain=1#L78
* https://github.com/open-telemetry/opentelemetry-specification/blob/3d60131181a6d4a82e1b7684c609e8528e86d9d2/specification/metrics/README.md?plain=1#L104
* https://github.com/open-telemetry/opentelemetry-specification/blob/3d60131181a6d4a82e1b7684c609e8528e86d9d2/specification/logs/data-model.md?plain=1#L476
* https://github.com/open-telemetry/opentelemetry-specification/blob/3d60131181a6d4a82e1b7684c609e8528e86d9d2/experimental/trace/zpages.md?plain=1#L27
* https://github.com/open-telemetry/opentelemetry-specification/blob/3d60131181a6d4a82e1b7684c609e8528e86d9d2/specification/configuration/file-configuration.md?plain=1#L120

**Comment by dyladan** [comment]
I really like this idea. I believe it may also help the PR review process when new specification is added if it explicitly calls out the OTEP which motivates it directly in the text.

**Comment by hterik** [comment]
This sounds like good idea for easier traceability. One has to be careful to not rely too heavily on it though, linking to xEPs should not be a substitute for good documentation. 

For an example of what can happen, see Python the typing module. Where the documentation was for a long time lacking and often referring back to old PEPs, that by time became superseded by later improvements and instead provided out-of-date examples. https://github.com/python/cpython/issues/91533

**Comment by reyang** [comment]
We will have a discussion in the upcoming TC meeting next year (2024).

**Comment by austinlparker** [comment]
We don't have a project to assign this to yet, but this is on the GC to solve.

**Comment by tigrannajaryan** [comment]
OTEPs 0035, 0099, 0122 are linked from https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md

Marking as done.

**Comment by tigrannajaryan** [comment]
We are looking for volunteers to edit the spec and add the references to OTEPs.

**Comment by reyang** [comment]
I've updated the metrics related OTEPs.
